stl/inc/filesystem

```diff
@@ -1377,6 +1377,36 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Handle the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
+                // !has_root_name(_Right) && !has_root_directory(_Right)
+                const auto _Left_size  = _Left._Text.size();
+                const auto _Left_first = _Left._Text.data();
+                const auto _Left_last  = _Left._Text.data() + _Left_size;
+
+                // Appending a slash to "X:" would've made it an absolute path
+                const bool _Left_is_just_drive = _Left_size == 2 && _Is_drive_prefix(_Left_first);
+                const bool _Is_slash_needed    = _Left_size != 0 && !_Left_is_just_drive && !_Is_slash(_Left_last[-1]);
+
+                const auto _Total_size = _Left_size + _Is_slash_needed + _Right_size;
+
+                path _Tmp;
+                _Tmp._Text._Resize_and_overwrite(_Total_size, [&](wchar_t* _Ptr, size_t _Size) {
+                    _CSTD memcpy(_Ptr, _Left_first, _Left_size * sizeof(wchar_t));
+                    _Ptr += _Left_size;
+                    if (_Is_slash_needed) {
+                        *_Ptr++ = preferred_separator;
+                    }
+                    _CSTD memcpy(_Ptr, _Right_first, _Right_size * sizeof(wchar_t));
+                    return _Size;
+                });
+                return _Tmp;
+            }

```

Comment:
FWIW, I agree that using `_Resize_and_overwrite` complicates the code. I suspect it creates a very happy hot path, though. Do you feel like it's worth benchmarking to determine if we should go back to the previous, simpler implementation?

Reply:
I'm willing to keep the somewhat more complicated resize-and-overwrite code without putting in the effort to benchmark it. This is a classic scenario where string copying dominates performance and unnecessary initialization nearly doubles the cost, and I can believe that paths can be quite long (a hundred characters easy, more if actual long path support is enabled).